pyvcloud/vcd/external_network.py

```diff
@@ -315,3 +315,42 @@ def __create_vimobj_ref(self, vc_href, pg_moref, pg_type):
         vim_object_ref.append(E_VMEXT.VimObjectType(pg_type))
 
         return vim_object_ref
+
+    def detach_port_group(self, vim_server_name, port_group_name):
+        """Detach a portgroup from an external network.
+
+        :param str vc_name: name of vc where portgroup is present.
+        :param str pg_name: name of the portgroup to be detached from
+             external network.
+
+        return: object containing vmext:VMWExternalNetwork XML element that
+             representing the external network.
+        :rtype: lxml.objectify.ObjectifiedElement
+        """
+        ext_net = self.get_resource()
+        platform = Platform(self.client)
+
+        if not vim_server_name or not port_group_name:
+            raise InvalidParameterException(
+                "Either vCenter Server name is none or portgroup name is none")
+
+        vc_record = platform.get_vcenter(vim_server_name)
+        vc_href = vc_record.get('href')
+        pg_moref_types = \
+            platform.get_port_group_moref_types(vim_server_name,
+                                                port_group_name)
+
+        vim_port_group_refs = \
+            ext_net['{' + NSMAP['vmext'] + '}VimPortGroupRefs']
```

Comment:
Need to handle the condition when the single port group is present that is "VimPortGroupRef".

Check in the VCD UI when single port group is present and you are trying to delete the same.